### DISCUSSION TITLE
Create RPG Procedure from Java Class/Methods

### DISCUSSION THREAD

**Discussion by thomprl** [codefori/vscode-ibmi#2282] [Ideas]
This is just an idea and probably not a lot of interest to others.  In RDi you can create a RPG Procedure from a Java JAR file/Class/Method.  

You basically locate the JAR file, then the package, then the Class.  One limitation of RDi is it only allows you to pick one method at a time.  It also imports any dependencies which is also kind of annoying if you have multiple methods as it imports duplicate methods and it has to be cleaned up.  

![image](https://github.com/user-attachments/assets/3cbac6c9-80ee-4772-855d-42a78ab8c906)

![image](https://github.com/user-attachments/assets/09e00c7e-706e-4fc3-8d2b-33d6afdd01be)

![image](https://github.com/user-attachments/assets/8e7d31ee-b01b-4297-9a3c-c7f493b30855)

**Comment by sebjulliand** [comment]
Sounds like a good idea to me!
It would not fit in the RPGLE extension I guess, just because we may not want to include a JVM with it to be able to dig into Java's bytecode to extract the classes information - but it would make for a cool side project (or specific extension). We could rely on an existing Java installation though... Let's keep it in mind 😉
